src/thumbnailer_utils.cc

```diff
@@ -54,72 +55,73 @@ UtilsStatus AnimData2Pictures(WebPData* const webp_data,
   return kOk;
 }
 
-UtilsStatus AnimData2PSNR(const std::vector<EnclosedWebPPicture>& original_pics,
+UtilsStatus AnimData2PSNR(const std::vector<Frame>& original_frames,
                           WebPData* const webp_data,
                           ThumbnailStatsPSNR* const stats) {
   if (stats == nullptr) return kMemoryError;
 
-  std::vector<EnclosedWebPPicture> pics;
-  const UtilsStatus data2pics_status = AnimData2Pictures(webp_data, &pics);
-  if (data2pics_status != kOk) return data2pics_status;
-
-  if (pics.size() != original_pics.size()) {
-    std::cerr << "Picture count mismatched.\n";
-    return kGenericError;
-  }
-
-  const int pic_count = original_pics.size();
+  std::vector<Frame> new_frames;
+  const UtilsStatus data2frames_status =
+      AnimData2Frames(webp_data, &new_frames);
+  if (data2frames_status != kOk) return data2frames_status;
+
+  const int frame_count = original_frames.size();
+  int new_frame_index = 0;
+  for (int i = 0; i < frame_count; ++i) {
+    const Frame& original_frame = original_frames[i];
+
+    // Check if the next frame of new_frames matches original_frame,
+    // based on their timestamps. This is because consecutive original frames
+    // having the exact same WebPPicture are merged into one.
+    if (new_frame_index + 1 < new_frames.size() &&
+        new_frames[new_frame_index + 1].timestamp == original_frame.timestamp) {
+      ++new_frame_index;
+    }
```

Comment:
You should check or assert than the timestamps match before calling `WebPPictureDistortion()`.